src/Features/Core/Portable/AddConstructorParametersFromMembers/State.cs

```diff
@@ -38,29 +31,75 @@ private class State
 
             private bool TryInitialize(
                 AddConstructorParametersFromMembersCodeRefactoringProvider service,
-                Document document,
-                TextSpan textSpan,
-                ImmutableArray<ISymbol> selectedMembers,
-                CancellationToken cancellationToken)
+                ImmutableArray<ISymbol> selectedMembers)
             {
                 if (!selectedMembers.All(IsWritableInstanceFieldOrProperty))
                 {
                     return false;
                 }
 
-                this.TextSpan = textSpan;
-                this.SelectedMembers = selectedMembers;
-                this.ContainingType = this.SelectedMembers[0].ContainingType;
+                this.ContainingType = selectedMembers[0].ContainingType;
                 if (this.ContainingType == null || this.ContainingType.TypeKind == TypeKind.Interface)
                 {
                     return false;
                 }
 
-                this.Parameters = service.DetermineParameters(selectedMembers);
+                var parameters = service.DetermineParameters(selectedMembers);
+                // We are trying to add these parameters into an existing constructor's parameter list.
+                // Comparing parameters based on names to make sure parameter list won't contains duplicate parameters after we
+                // append the new parameters
+                this.ConstructorToAddTo = GetDelegatedConstructorBasedOnParameterNames(this.ContainingType, parameters);
 
-                this.MatchingConstructor = service.GetMatchingConstructor(this.ContainingType, this.Parameters);
-                this.DelegatedConstructor = service.GetDelegatedConstructor(this.ContainingType, this.Parameters);
-                return this.DelegatedConstructor != null;
+                if (this.ConstructorToAddTo == null)
+                {
+                    return false;
+                }
+
+                var zippedParametersAndSelectedMembers = parameters.Zip(selectedMembers, (parameter, selectedMember) => (parameter, selectedMember));
+                var missingParametersBuilder = ArrayBuilder<IParameterSymbol>.GetInstance();
+                var missingMembersBuilder = ArrayBuilder<ISymbol>.GetInstance();
+                var constructorParamNames = this.ConstructorToAddTo.Parameters.SelectAsArray(p => p.Name);
+                foreach ((var parameter, var selectedMember) in zippedParametersAndSelectedMembers)
+                {
+                    if (!constructorParamNames.Contains(parameter.Name))
+                    {
+                        missingParametersBuilder.Add(parameter);
+                        missingMembersBuilder.Add(selectedMember);
+                    }
+                }
+
+                this.MissingParameters = missingParametersBuilder.ToImmutableAndFree();
+                this.MissingMembers = missingMembersBuilder.ToImmutableAndFree();
+                return true;
```

Comment:
dumb question.  should we only be returning true if missingparams/missingmembers have at least one element in them?

Reply:
Emmm, this 'ConstructorToAdd' is already checked it has 'more parameters' than the missing parameter, so here 'MissingParameters' and 'MissingMembers' must contain something.

But it doesn't hurt to check it :)